src/_guides/language/evolution.md

```diff
@@ -269,10 +269,82 @@ except within the `@dart` and version strings.
 As the example above shows,
 other comments can appear before the `@dart` comment.
 
-
 For more information about how language versioning works, see the
 [language versioning specification][language versioning feature].
 
+
+## Mixed-mode programs
+
+You can mix language versions within an app,
+as long as the major numbers are the same. **_[PENDING: check!]_**
+In the context of null safety,
+an app can use some libraries that
+are null safe and some that aren't.
+These apps are called **mixed-mode programs.**
+
+{{ site.alert.info }}
+  Except for a few edge cases,
+  a mixed-mode program should do exactly what you expect.
+  One example of an edge case is an `is` test
+  on an instance of a generic type from a closure
+  that crosses the boundary between null-safe and null-unsafe libraries.
+  **_[PENDING: is there a list of edge cases that we can point to?]_**
+{{ site.alert.end }}
+
+Mixed mode lets you migrate to new features at your own pace.
+It also frees package maintainers to migrate their code,
+with the knowledge that even legacy users can get new
+bug fixes and other improvements.
+However, mixed-mode programs don't get all the advantages
+that features such as null safety can bring.
+
+**_[PENDING: Is it OK to talk about mixed mode generically,
+or is it tied tightly to null safety?]_**
```

Comment:
Yes, I'm generally in favor of 1) tying mixed version docs with null safety, and 2) considering moving the mixed version + manual migration content out into a separate advanced topic so that this page can focus on the recommended flow (migrating in order).

Reply:
The mixed version + detailed manual migration content are in #2764.